Main.py

```diff
@@ -1,82 +1,102 @@
                     simEnv['next_divition'][lagging] += MINIMUM_TIME_RESOLUTION
-                    print lagging
 
                 ## Divide
                 toDivide = (simEnv['next_divition'][0:nr_alive] <= t)#.dot(simEnv['lag_time'][0:nr_alive] >= t)
                 nr_Divide = np.sum(toDivide)
                 if nr_Divide != 0:
-                    #print nr_Divide
-                    #print np.nonzero(toDivide)[0]
+                    # print nr_Divide
+                    # print np.nonzero(toDivide)[0]
                     idxDivide = np.nonzero(toDivide)[0]
                     toBirth = np.arange(nr_alive,(nr_alive + nr_Divide))
                     simEnv['founder_id'][toBirth] = idxDivide
                     simEnv['lag_time'][toBirth] = simEnv['lag_time'][toDivide]
                     simEnv['cell_cycle_time'][toBirth] = simEnv['cell_cycle_time'][toDivide]
                     simEnv['next_divition'][toDivide] = t + simEnv['cell_cycle_time'][toDivide]
                     simEnv['next_divition'][toBirth] = simEnv['next_divition'][toDivide]
-                    #CELL_CYCLE_TIMES[simEnv['founder_id'][toDivide]]
+                    # CELL_CYCLE_TIMES[simEnv['founder_id'][toDivide]]
                     simEnv['nr_divitions'][toDivide] += 1
 
                     ## Mutate
                     toMutate = np.random.random(nr_Divide) < PROB_MUTATION
                     if any(toMutate):
-                        mutateORF = np.random.random([sum(toMutate),1]) * (data1_Cum[-1]) # pick mutation
+                        mutateORF = np.random.random([sum(toMutate), 1]) * (data1_Cum[-1]) # pick mutation
                         idxORF = np.searchsorted(data1_Cum, mutateORF, side='right') # peek mutation
                         mutation[toBirth[toMutate]] = idxORF  # store mutation
-                        #idxDivide = np.nonzero(toDivide)[0]
-                        #nr_mutations +=
+                        # idxDivide = np.nonzero(toDivide)[0]
+                        # nr_mutations +=
                         print 'time is ', t
                         print 'alive: ', nr_alive
-                        print data4[idxORF,iEnv]
-
+                        print data4[idxORF, iEnv]
                         simEnv['cell_cycle_time'][toBirth[toMutate]] = simEnv['cell_cycle_time'][idxDivide[toMutate]] + \
-                                                                       data4[idxORF,iEnv]*MEAN_CELL_CYCLE_TIME #(CELL_CYCLE_TIMES[simEnv['founder_id'][idxDivide[toMutate]]])
+                                                                       data4[idxORF, iEnv]*MEAN_CELL_CYCLE_TIME #(CELL_CYCLE_TIMES[simEnv['founder_id'][idxDivide[toMutate]]])
                         simEnv['next_divition'][toBirth[toMutate]] = t + simEnv['cell_cycle_time'][toBirth[toMutate]]
 
                 ## Population ages
                 simEnv['age'][0:nr_alive] += 0.1 # worry, might take up alotta time.
```

Comment:
> I don't understand what speed up you are referring to, this is a structured array, not a dictionary. See:

I didn't imply that this was a speed-up, just that it actually doesn't take too much time. However each time you do `simEnv['age']` you create a python object called a numpy-view or something similar. So you could avoid that by creating the view once before all cycles and then update the view. This will not be a great time-save but assuming you check in on the simulation 200 times per cycle and do 50 cycles you might save a second.

More importantly, many numpy-functions have an `out`-parameter, and this implies that the data will be placed directly inside the `out` array instead of creating a new array and then assigning the values back into your array. Whenever this is possible, I would say you need to use it.